pkg/services/anonymous/anonimpl/impl.go

```diff
@@ -96,5 +120,26 @@ func (a *AnonDeviceService) TagDevice(ctx context.Context, httpReq *http.Request
 
 	a.localCache.SetDefault(key, struct{}{})
 
-	return a.remoteCache.Set(ctx, key, []byte(key), thirtyDays)
+	deviceJSON, err := json.Marshal(taggedDevice)
+	if err != nil {
+		return err
+	}
+
+	if err := a.remoteCache.Set(ctx, key, deviceJSON, thirtyDays); err != nil {
+		return err
+	}
+
+	untagKind := anonymous.AnonDevice
```

Comment:
should we move this logic into the untagDevice method?

if not; could we add a comment as to why this is needed in the TagDevice piece

Reply:
I'd avoid to keep the primitive `untagDevice` without inverse behavior (you pass as an arg a Device but actually what get's untagged is the inverse doesn't seem straightforward)